docs/snippets.md

```diff
@@ -94,11 +94,27 @@ add_action( 'plugins_loaded', function() {
  *
  * This filter is used to filter the arguments sent to the remote server during a push. The below code snippet passes the original published date to the new pushed post and sets the same published date instead of setting it as per the current time.
  */
-add_filter( 'dt_push_post_args', function( $post_body, $post ) {
-    $post_body['post_date'] = $post->post_date;
+add_filter( 'dt_push_post_args', function( $post_body, $post, $args, $connection ) {
+
+    // When pushing to an external connection, we use the REST API, so the name of the field is `date`.
+    // But when pushing to an internal connection, the attributes are sent to wp_insert_post, which expects `post_date`.
+    $field_prefix = is_a( $connection, 'Distributor\ExternalConnections\WordPressExternalConnection' ) ? '' : 'post_';
```

Comment:
```suggestion
    $field_prefix =( $connection instanceof \Distributor\ExternalConnections\WordPressExternalConnection ) ? '' : 'post_';
```

This is a bit of a nitpick but [PHP have tried to deprecate `is_a()` a couple of times](https://github.com/PHPCSStandards/PHPCSExtra/issues/260) so it's nice to avoid it in case another attempt is made.

Reply:
BTW: I know I gave you the bad steer on this -- sorry about that. I had a dumb moment.